pyccel/ast/numpyext.py

```diff
@@ -705,24 +704,64 @@ def is_elemental(self):
 #==============================================================================
 class NumpyWhere(PyccelInternalFunction):
     """ Represents a call to  numpy.where """
-    __slots__ = ()
+
+    __slots__ = ('_condition', '_x', '_y', '_dtype', '_rank', '_shape', '_order', '_precision')
+    _attribute_nodes = ('_condition','_x','_y')
     name = 'where'
 
-    def __init__(self, mask):
-        super().__init__(mask)
+    def __init__(self, condition, x, y):
+        self._condition = condition
+        self._x = x
+        self._y = y
+
+        args      = (x, y)
+        integers  = [e for e in args if e.dtype is NativeInteger() or e.dtype is NativeBool()]
+        floats     = [e for e in args if e.dtype is NativeFloat()]
+        complexs  = [e for e in args if e.dtype is NativeComplex()]
+
+        if complexs:
+            self._dtype     = NativeComplex()
+            self._precision = max(e.precision for e in complexs)
+        elif floats:
+            self._dtype     = NativeFloat()
+            self._precision = max(e.precision for e in floats)
+        elif integers:
+            self._dtype     = NativeInteger()
+            self._precision = max(e.precision for e in integers)
+        else:
+            raise TypeError('cannot determine the type of {}'.format(self))
+
+        shape = broadcast(x._shape, y._shape)
+        shape = broadcast(condition._shape, shape)
 
+        self._rank = len(shape)
+        self._shape = shape
+        self._order = 'C'
+        super().__init__(condition, x, y)
 
     @property
-    def mask(self):
-        return self._args[0]
+    def condition(self):
+        """When True, yield x, otherwise yield y."""
```

Comment:
I don't think this is a very good description of what this function does. As written I expect either `x` or `y` to be returned from this function